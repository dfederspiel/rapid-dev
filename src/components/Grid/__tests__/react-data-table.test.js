import React from 'react';
import ReactDataTable from '../react-data-table';
import renderer from 'react-test-renderer';

describe('the react data table component', () => {

    const props = {
        data: [
            { action: '', test:'hello'},
            { action: '', test:'world'}
        ],
        columns: [
            {
                name: 'Action',
                selector: 'action',
                cell: row => <input type="button" value="View" />,
                maxWidth: '20px'
            },
            {
                name: 'Test',
                selector: 'test',
                sortable: true
            }
        ]
    };

    it('renders correctly', () => {
        const tree = renderer.create(
            <ReactDataTable
                {...props}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});