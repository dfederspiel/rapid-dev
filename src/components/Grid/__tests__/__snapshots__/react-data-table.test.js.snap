// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the react data table component renders correctly 1`] = `
<div
  className="react-data-table sc-ckVGcZ gVZomo"
  style={Object {}}
>
  <div
    className="sc-eNQAEJ fWmJgw"
  >
    <div
      className="rdt_Table sc-bdVaJa bTnnmE"
      disabled={false}
    >
      <div
        className="rdt_TableHead sc-bwzfXH gXdTHx"
      >
        <div
          className="rdt_TableHeadRow sc-bxivhb bojDWV"
        >
          <div
            className="sc-ifAKCX sc-EHOje rdt_TableCol sc-VigVT dwYePf"
            id="column-action"
            onClick={[Function]}
          >
            <div
              className="sc-jTzLTM frKpZh"
            >
              Action
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje rdt_TableCol sc-VigVT jCkXQs"
            id="column-test"
            onClick={[Function]}
          >
            <div
              className="sc-jTzLTM frKpZh"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="rdt_TableBody sc-dxgOiQ gXekDi"
        offset="250px"
      >
        <div
          className="rdt_TableRow sc-gqjmRU dGyVpY"
          onClick={[Function]}
        >
          <div
            className="sc-ifAKCX sc-EHOje rdt_TableCell sc-bZQynM eLFFxh"
          >
            <div
              className="sc-gzVnrw jltdJX"
              data-tag="___react-data-table--click-clip___"
            />
            <div
              className="react-data-table--cell-content"
            >
              <input
                type="button"
                value="View"
              />
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje rdt_TableCell sc-bZQynM cgAqRg"
          >
            <div
              className="sc-gzVnrw jltdJX"
              data-tag="___react-data-table--click-clip___"
            />
            <div
              className="react-data-table--cell-content"
            >
              hello
            </div>
          </div>
        </div>
        <div
          className="rdt_TableRow sc-gqjmRU dGyVpY"
          onClick={[Function]}
        >
          <div
            className="sc-ifAKCX sc-EHOje rdt_TableCell sc-bZQynM eLFFxh"
          >
            <div
              className="sc-gzVnrw jltdJX"
              data-tag="___react-data-table--click-clip___"
            />
            <div
              className="react-data-table--cell-content"
            >
              <input
                type="button"
                value="View"
              />
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje rdt_TableCell sc-bZQynM cgAqRg"
          >
            <div
              className="sc-gzVnrw jltdJX"
              data-tag="___react-data-table--click-clip___"
            />
            <div
              className="react-data-table--cell-content"
            >
              world
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
